tests/Framework.py

```diff
@@ -418,14 +419,26 @@ def ignoreWarning(self, category=Warning, module=""):
             warnings.filterwarnings("ignore", category=category, module=module)
             yield
 
+    @contextlib.contextmanager
+    def replayData(self, filename: str):
+        previous = self.__customFilename
+        self.__customFilename = filename
+        try:
+            yield
+        finally:
+            self.__customFilename = previous
+
     def __openFile(self, mode):
-        for _, _, functionName, _ in traceback.extract_stack():
-            if functionName.startswith("test") or functionName == "setUp" or functionName == "tearDown":
-                if functionName != "test":  # because in class Hook(Framework.TestCase), method testTest calls Hook.test
-                    fileName = os.path.join(
-                        self.replayDataFolder,
-                        f"{self.__class__.__name__}.{functionName}.txt",
-                    )
+        fileName = None
+        if self.__customFilename:
+            fileName = self.__customFilename
+        else:
+            for _, _, functionName, _ in traceback.extract_stack():
+                if functionName.startswith("test") or functionName == "setUp" or functionName == "tearDown":
+                    # because in class Hook(Framework.TestCase), method testTest calls Hook.test
+                    if functionName != "test":
+                        fileName = f"{self.__class__.__name__}.{functionName}.txt"
+        fileName = os.path.join(self.replayDataFolder, fileName) if fileName else None
```

Comment:
>_possible bug_

**✨ AI Suggestion:** The variable `fileName` might be `None` if `self.__customFilename` is not set and the loop over `traceback.extract_stack()` does not find a suitable function name. Calling `os.path.join(self.replayDataFolder, None)` will raise a `TypeError`. Add a check to ensure `fileName` is not `None` before joining the path.
```suggestion
                        fileName = f"{self.__class__.__name__}.{functionName}.txt"
        fileName = os.path.join(self.replayDataFolder, fileName) if fileName is not None else None
```